### PR TITLE
Update to Weld 5.0.0.Beta1

### LINKED ISSUE #19910: Update to Weld 5.0.0.Beta1 and CDI API 4.0.0-RC3
https://repo1.maven.org/maven2/org/jboss/weld/weld-osgi-bundle/5.0.0.Beta1/weld-osgi-bundle-5.0.0.Beta1.jar

https://repo1.maven.org/maven2/org/jboss/weld/se/weld-se-core/5.0.0.Beta1/weld-se-core-5.0.0.Beta1.jar

https://repo1.maven.org/maven2/jakarta/enterprise/jakarta.enterprise.cdi-api/4.0.0.Beta3/jakarta.enterprise.cdi-api-4.0.0.Beta3.jar

### PR DESCRIPTION
**Pull request by tevans78**
Various CDI 4.0 updates;
- jakarta.enterprise:jakarta.enterprise.cdi-api:4.0.0-RC3
- jakarta.enterprise:jakarta.enterprise.lang-model:4.0.0-RC3
- jakarta.interceptor:jakarta.interceptor-api:2.1.0-RC3
- org.jboss.weld.se:weld-se-core:5.0.0.Beta1
- org.jboss.weld:weld-osgi-bundle:5.0.0.Beta1

#build

### CODE REVIEW

File: dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd [hunk 1]
```diff
@@ -11,38 +11,44 @@
...
- jakarta.enterprise.event; version=3.0, \
- jakarta.enterprise.inject; version=3.0, \
- jakarta.enterprise.inject.literal; version=3.0, \
- jakarta.enterprise.inject.spi; version=3.0, \
- jakarta.enterprise.inject.spi.configurator; version=3.0, \
- jakarta.enterprise.util; version=3.0, \
- jakarta.inject; version=2.0, \
- jakarta.interceptor;version=2.0, \
+Import-Package: \
+ jakarta.el;version="4.0",\
```
Comment (jhanders34) [inline dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd:42]: Shouldn't the import be a range?

File: dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd [hunk 2]
```diff
@@ -51,15 +57,13 @@ Include-Resource: \
...
 instrument.disabled: true
 
 publish.wlp.jar.suffix: dev/api/spec
 
 -buildpath: \
-	jakarta.inject:jakarta.inject-api;version=2.0.1,\
-	jakarta.enterprise:jakarta.enterprise.cdi-api;version=4.0.0,\
+	jakarta.inject:jakarta.inject-api;version="[2.0.1,3)",\
+	jakarta.enterprise:jakarta.enterprise.cdi-api;version="[4.0,5)",\
+	jakarta.enterprise:jakarta.enterprise.lang-model;version="[4.0,5)",\
```
Comment (jhanders34) [inline dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd]: I have not seen ranges used in a build path before.  Does this work?

File: dev/io.openliberty.jakarta.interceptor.2.1/.classpath [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
```
Comment (jhanders34) [inline dev/io.openliberty.jakarta.interceptor.2.1/.classpath]: This should be Java 11

File: dev/io.openliberty.jakarta.interceptor.2.1/bnd.bnd [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#*******************************************************************************
+# Copyright (c) 2020, 2022 IBM Corporation and others.
```
Comment (jhanders34) [inline dev/io.openliberty.jakarta.interceptor.2.1/bnd.bnd:2]: Shouldn't it just be 2022?

File: dev/io.openliberty.jakarta.interceptor.2.1/.settings/org.eclipse.jdt.core.prefs [hunk 5]
```diff
@@ -0,0 +1,276 @@
+#Ant properties
+#Automatically generated by the ant prepare.settings.files task
+org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=enabled
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.8
+org.eclipse.jdt.core.compiler.compliance=1.8
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
Comment (jhanders34) [inline dev/io.openliberty.jakarta.interceptor.2.1/.settings/org.eclipse.jdt.core.prefs]: The targetPlatform, compliance and source should all be 11

File: dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd [hunk 6]
```diff
@@ -11,38 +11,44 @@
 -include= ~../cnf/resources/bnd/bundle.props
 bVersion=1.0
```
Comment (jhanders34) [inline dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd:13]: Should have:
```
javac.source: 11
javac.target: 11

Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=11))"
```
added to the bnd file and then update the .classpath and org.eclipse.jdt.core.prefs to have 11 instead of 1.8 in its java settings.

File: dev/io.openliberty.jakarta.interceptor.2.1/bnd.bnd [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props
+bVersion=1.0
+
+Bundle-SymbolicName: io.openliberty.jakarta.interceptor.2.1; singleton:=true
+
```
Comment (jhanders34) [inline dev/io.openliberty.jakarta.interceptor.2.1/bnd.bnd:18]: Should have:
```
javac.source: 11
javac.target: 11

Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=11))"
```
added to the bnd file.

File: dev/io.openliberty.jakarta.interceptor.2.1/bnd.bnd [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+Export-Package: jakarta.interceptor;version="2.1.0"
+
+instrument.disabled: true
+
+publish.wlp.jar.suffix: dev/api/spec
+
+-buildpath: \
+	jakarta.interceptor:jakarta.interceptor-api;version="[2.1,3)"
```
Comment (jhanders34) [inline dev/io.openliberty.jakarta.interceptor.2.1/bnd.bnd]: Again weird seeing a range in a buildpath.

File: dev/io.openliberty.org.jboss.weld5/bnd.overrides [hunk 9]
```diff
@@ -7,23 +7,194 @@ bFullVersion=${version;==;${bVersion}}.${libertyBundleMicroVersion}
...
+ org.jboss.weld.security;version="5.0.0.Beta1";uses:="org.jboss.weld.security.spi",\
+ org.jboss.weld.util;version="5.0.0.Beta1";uses:="jakarta.el,jakarta.enterprise.context.spi,jakarta.enterprise.event,jakarta.enterprise.inject,jakarta.enterprise.inject.spi,jakarta.enterprise.util,jakarta.interceptor,org.jboss.weld.bean,org.jboss.weld.bootstrap,org.jboss.weld.bootstrap.api,org.jboss.weld.bootstrap.spi,org.jboss.weld.exceptions,org.jboss.weld.inject,org.jboss.weld.injection,org.jboss.weld.interceptor.spi.model,org.jboss.weld.manager,org.jboss.weld.resources.spi,org.jboss.weld.serialization.spi",\
+ org.jboss.weld.xml;version="5.0.0.Beta1";uses:="org.jboss.weld.bootstrap.spi,org.xml.sax,org.xml.sax.helpers"
 
+# The first 2 imported packages are additions, the rest come from the original manifest
 Import-Package: \
  javax.security.auth,\
  jakarta.xml.ws;version="3.0",\
```
Comment (jhanders34) [inline dev/io.openliberty.org.jboss.weld5/bnd.overrides]: Should this be a range instead of just 3.0

**Review by jhanders34 — CHANGES_REQUESTED** [review]

File: dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd (hunk 2 above)
Comment (Azquelt) [reply dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd]: Yes. ~~Also, if you just put a number, it's automatically an unbound range unless you also put `EXACT`.~~

Having a range is nice because it means that we just have to update oss_dependencies.maven to consume a new release.

However, for these build path entries, I think we should only allow a micro version increase. If we consume a CDI API 4.0.1 release, we'd want that in this bundle. However, if there's a CDI API 4.1, that would be a new spec version and it would go in its own bundle so we wouldn't our range here to include it.

File: dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd (hunk 1 above)
Comment (tevans78) [reply dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd:42]: I don't know ... I copied most of these imports and exports from the original bundles. The original `jakarta.enterprise.cdi-api` manifest does not use a range for this so neither have I.

File: dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd (hunk 2 above)
Comment (tevans78) [reply dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd]: Yes, however we discussed in our squad and decided that the range needs to be tighter e.g. `"[4.0,4.1)"`

File: dev/io.openliberty.jakarta.interceptor.2.1/bnd.bnd (hunk 4 above)
Comment (tevans78) [reply dev/io.openliberty.jakarta.interceptor.2.1/bnd.bnd:2]: Well grey area ... the file was originally copied from the 2.0 version and then modified for 2.2 ... so technically I think the dates are correct

File: dev/io.openliberty.jakarta.interceptor.2.1/bnd.bnd (hunk 8 above)
Comment (tevans78) [reply dev/io.openliberty.jakarta.interceptor.2.1/bnd.bnd]: Again, we discussed and decided that a range is fine but will make it tighter `"[2.1,2.2)"`

File: dev/io.openliberty.org.jboss.weld5/bnd.overrides (hunk 9 above)
Comment (tevans78) [reply dev/io.openliberty.org.jboss.weld5/bnd.overrides]: Hmm now you mention it, I don't know why that import is there. It has clearly just been copied from the previous version. I understand why an import for `javax.security.auth` is needed but not `jakarta.xml.ws`???

**Comment by tevans78** [comment]
#build

File: dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd (hunk 2 above)
Comment (jhanders34) [reply dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd]: As long as the 3.0 bundles do start picking up the 4.0 since those are unbounded.  We may want to improve these ranges in a lot of the files to avoid older versions picking up new ones with the unbounded version and not specifying exact.

**Review by jhanders34 — APPROVED** [review]

File: dev/com.ibm.websphere.appserver.features/visibility/public/cdi-4.0/io.openliberty.cdi-4.0.feature [hunk 10]
```diff
@@ -65,7 +69,7 @@ Subsystem-Name: Jakarta Contexts and Dependency Injection 4.0
  io.openliberty.cdi.4.0.internal.weld, \
  com.ibm.ws.cdi.interfaces.jakarta, \
  io.openliberty.cdi.spi; location:="dev/spi/ibm/,lib/"
--jars=io.openliberty.cdi.4.0.thirdparty; location:="dev/api/third-party/,lib/"; mavenCoordinates="org.jboss.weld:weld-osgi-bundle:5.0.0.Alpha2"
+-jars=io.openliberty.cdi.4.0.thirdparty; location:="dev/api/third-party/,lib/"; mavenCoordinates="org.jboss.weld:weld-osgi-bundle:5.0.0"
```
Comment (Emily-Jiang) [inline dev/com.ibm.websphere.appserver.features/visibility/public/cdi-4.0/io.openliberty.cdi-4.0.feature]: This maven coordinate says 5.0.0 instead of 5.0.0.Beta1?

File: dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd (hunk 2 above)
Comment (Azquelt) [reply dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd]: Just had a check through the personal build, it's definitely packaging the 3.0 jar correctly (class file modified dates are different and classes added to existing packages in 4.0 are not present) and that just lists a version number rather than a range, so I must be mistaken. We should check exactly what it does do.

Possibly I was confusing it with other bundles where we pull in the contents using `Include-Resource: @${repo;artifact:version}` where we do seem to consistently use ranges.

File: dev/com.ibm.websphere.appserver.features/visibility/public/cdi-4.0/io.openliberty.cdi-4.0.feature (hunk 10 above)
Comment (tevans78) [reply dev/com.ibm.websphere.appserver.features/visibility/public/cdi-4.0/io.openliberty.cdi-4.0.feature]: I don't think that version number matters for now ... the whole thing is a bit questionable because that jar really isn't the same as the bundle you would find at those maven co-ordinates! I have no idea what the correct GAV should be here.

Comment (Azquelt) [reply dev/com.ibm.websphere.appserver.features/visibility/public/cdi-4.0/io.openliberty.cdi-4.0.feature]: >that jar really isn't the same as the bundle you would find at those maven co-ordinates!

I think that's potentially a problem. IIRC the liberty maven and gradle plugins use this to add the right API dependencies when the user specifies dependencies on particular features.

File: dev/cnf/oss_dependencies.maven [hunk 11]
```diff
@@ -141,9 +141,11 @@ jakarta.batch:jakarta.batch-api:2.0.0
 jakarta.ejb:jakarta.ejb-api:4.0.0
 jakarta.enterprise.concurrent:jakarta.enterprise.concurrent-api:2.0.0
 jakarta.enterprise:jakarta.enterprise.cdi-api:3.0.0
-jakarta.enterprise:jakarta.enterprise.cdi-api:4.0.0.Beta3
+jakarta.enterprise:jakarta.enterprise.cdi-api:4.0.0-RC3
+jakarta.enterprise:jakarta.enterprise.lang-model:4.0.0-RC3
```
Comment (Emily-Jiang) [inline dev/cnf/oss_dependencies.maven:145]: Why did you add lang-mode separately?

File: dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd [hunk 12]
```diff
@@ -11,55 +11,66 @@
...
+ jakarta.enterprise.inject.build.compatible.spi;version="3.0";uses:="jakarta.enterprise.context.spi,jakarta.enterprise.event,jakarta.enterprise.inject,jakarta.enterprise.inject.spi,jakarta.enterprise.lang.model,jakarta.enterprise.lang.model.declarations,jakarta.enterprise.lang.model.types",\
+ jakarta.enterprise.inject.literal;version="3.0";uses:="jakarta.enterprise.util,jakarta.inject",\
+ jakarta.enterprise.inject.se;version="3.0";uses:="jakarta.enterprise.inject,jakarta.enterprise.inject.spi",\
+ jakarta.enterprise.inject.spi;version="3.0";uses:="jakarta.el,jakarta.enterprise.context.spi,jakarta.enterprise.event,jakarta.enterprise.inject,jakarta.enterprise.inject.spi.configurator,jakarta.interceptor",\
+ jakarta.enterprise.inject.spi.configurator;version="3.0";uses:="jakarta.enterprise.context.spi,jakarta.enterprise.event,jakarta.enterprise.inject,jakarta.enterprise.inject.spi,jakarta.enterprise.util",\
+ jakarta.enterprise.util;version="3.0",\
+ jakarta.inject;version="2.0.1",\
+ jakarta.enterprise.lang.model;version="4.0";uses:="jakarta.enterprise.lang.model.declarations,jakarta.enterprise.lang.model.types",\
```
Comment (Emily-Jiang) [inline dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd:36]: The version should be 3.0 not 4.0

File: dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd [hunk 13]
```diff
@@ -11,55 +11,66 @@
...
+ jakarta.enterprise.lang.model.declarations;version="4.0";uses:="jakarta.enterprise.lang.model,jakarta.enterprise.lang.model.types",\
```
Comment (Emily-Jiang) [inline dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd:37]: same as above

File: dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd [hunk 14]
```diff
@@ -11,55 +11,66 @@
...
+ jakarta.enterprise.lang.model.types;version="4.0";uses:="jakarta.enterprise.lang.model,jakarta.enterprise.lang.model.declarations"
```
Comment (Emily-Jiang) [inline dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd:38]: version should be 3.0

Comment (Emily-Jiang) [reply dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd:38]: I opened the jar and found all of the version are 3.0. I thought they would be 4.0, same as the bundle version, because they did not use semantic versioning policy.

File: dev/cnf/oss_dependencies.maven (hunk 11 above)
Comment (tevans78) [reply dev/cnf/oss_dependencies.maven:145]: because it is separate in maven central

File: dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd (hunk 14 above)
Comment (Emily-Jiang) [reply dev/io.openliberty.jakarta.cdi.4.0/bnd.bnd:38]: I will follow this up with Jakarta CDI spec.

**Review by Emily-Jiang — APPROVED** [review]

**Comment by tevans78** [comment]
This PR is ready to merge but will hold it out until after the GM. Although it has been tested, I don't want to risk breaking the concurrent-3.0 beta.

**Comment by tevans78** [comment]
#libby

**Comment by wraschke** [comment]
We need to revert this PR as it's caused build break like

`ERROR: io.openliberty.jakarta.interceptor-2.0
[MISSING_FEATURE io.openliberty.jakarta.interceptor-2.0]
Feature io.openliberty.jakarta.interceptor-2.0 1.0.0 is not found in the new build. Has the feature been deleted? or is the build incomplete?
For More Information, visit http://was.pok.ibm.com/xwiki/bin/view/Build/FeatureChecker`
